File: app/src/main/java/com/witype/Dragger/di/component/AppComponent.java
package com.witype.Dragger.di.component;

import android.app.Application;

import com.witype.Dragger.di.module.ClientModule;
import com.witype.Dragger.di.module.ConfigModule;
import com.witype.Dragger.integration.CallDataModel;
import com.witype.Dragger.integration.IRequestManager;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;

@Singleton
@Component(modules = {ClientModule.class, ConfigModule.class})
public interface AppComponent {

    void inject(Application application);

    IRequestManager getRequestManager();

    CallDataModel getCallDataModel();

    @Component.Builder
    interface Builder {

        @BindsInstance
        Builder application(Application application);

        Builder config(ConfigModule configModule);

        AppComponent build();
    }
}

File: mvp/src/main/java/com/witype/mvp/di/component/AppComponent.kt
package com.witype.mvp.di.component

import android.app.Application
import com.witype.mvp.di.module.ClientModule
import com.witype.mvp.di.module.ConfigModule
import com.witype.mvp.integration.IRequestManager
import com.witype.mvp.integration.RetrofitRequestManager
import dagger.BindsInstance
import dagger.Component
import retrofit2.Retrofit
import javax.inject.Singleton

@Singleton
@Component(modules = [ClientModule::class, ConfigModule::class])
interface AppComponent {

    fun inject(application: Application)

    fun IRequestManager() : RetrofitRequestManager

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: Application) : Builder

        fun config(configModule: ConfigModule) : Builder

        fun build() : AppComponent

    }
}